dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -316,33 +333,84 @@ public ResponseEntity<ResponseBodyEmitter> execute(
   }
 
   protected ResponseEntity<Flux<DataBuffer>> retrieve(
-      WebClient.RequestHeadersSpec<?> requestHeadersSpec) {
+      WebClient.RequestHeadersSpec<?> requestHeadersSpec,
+      HttpMethod httpMethod,
+      String upstreamUrl,
+      String routeId,
+      UserDetails userDetails) {
     WebClient.ResponseSpec responseSpec =
         requestHeadersSpec
             .retrieve()
             .onStatus(httpStatusCode -> true, clientResponse -> Mono.empty());
 
+    Audit.AuditBuilder auditBuilder =
```

Comment:
Are we building the audit even if the feature is disabled? Seems wasteful to me even if the entry is never processed later.